### ISSUE TITLE
About the “cca_loss”

### ISSUE THREAD

**Issue by YuKDseele** [LabShuHangGU/CCA#6]
Hello, I have a few questions regarding the **CCA loss**:

1. **Am I misunderstanding the CCA loss?**  
   The formula for CCA loss is derived by subtracting the bitrate estimated by the auxiliary entropy model from the major entropy model:  
   ```python
   out["cca_loss"] = (
       torch.log(output["likelihoods"]["y"]).sum() / (-math.log(2)) - 
       torch.log(output["aux_likelihoods"]["y_cca"]).sum() / (-math.log(2))
   ) / num_pixels
   ```
   According to Eq.6 and Eq.7 of the paper, the CCA loss is derived by subtracting the entropy of the major model from the auxiliary model. To emphasize the difference between the models, and since the auxiliary model’s entropy is usually larger, a negative sign is added, resulting in a negative CCA loss. Is this understanding correct?

2. **Is a CCA loss of -22.27282 during training problematic?**  
   If the CCA loss reflects the prediction gap between the models, a gap exceeding 22 bits might indicate problem. Given that the auxiliary model is autoregressive (with reduced dependencies), could this large gap suggest a problem in my training process?
   It indicates that the auxiliary entropy model predicts \(y\) with over 22 bits, which is unusual. I would like to know if this could be caused by the inappropriate parameter settings. If so, how can I solve this problem

**Comment by MichTheEncoder** [comment]
Actually, a counterintuitive point is that the estimated entropy of the auxiliary entropy model is optimized towards the **worse**, which leads to an inaccurate estimate and larger result. Thus the CCA loss is usually a large negative value.

**Comment by using369** [comment]
@YuKDseele Hello, the CCA loss is also very large (~-28) during my training, is it normal?

**Comment by JChenenene** [comment]
Hello, the CCA loss is also very large (~-37.009) during my training, is it normal?
